plugins/modules/ndo_mcp_global_policy.py

```diff
@@ -0,0 +1,323 @@
+    template: fabric_template
+    state: query
+  register: query_all
+
+- name: Delete the MCP Global Policy object with name
+  cisco.mso.ndo_mcp_global_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: fabric_template
+    name: example_mcp_global_policy
+    state: absent
+
+- name: Delete the MCP Global Policy object with UUID
+  cisco.mso.ndo_mcp_global_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: fabric_template
+    uuid: "{{ mcp_global_policy_all.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.constants import ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True, aliases=["fabric_template"]),
+        name=dict(type="str", aliases=["mcp_global_policy"]),
+        uuid=dict(type="str", aliases=["mcp_global_policy_uuid"]),
+        description=dict(type="str"),
+        admin_state=dict(type="str", choices=["enabled", "disabled"]),
+        key=dict(type="str", no_log=False),
+        per_vlan=dict(
+            type="str",
+            aliases=["per_epg", "mcp_pdu_per_vlan"],
+            choices=["enabled", "disabled"],
+        ),
+        loop_detection_factor=dict(type="int", aliases=["loop_factor", "loop_detection_mult_factor"]),
+        port_disable=dict(
+            type="str",
+            aliases=["port_disable_protection"],
+            choices=["enabled", "disabled"],
+        ),
+        initial_delay_time=dict(type="int", aliases=["initial_delay"]),
+        transmission_frequency_sec=dict(type="int", aliases=["tx_freq"]),
+        transmission_frequency_msec=dict(type="int", aliases=["tx_freq_ms"]),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    # Enforcing that a user must specify a name or uuid when
+    # adding, updating or removing even though there is only one policy per template.
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    state = module.params.get("state")
+    per_vlan = module.params.get("per_vlan")
+    port_disable = module.params.get("port_disable")
+
+    mso_values = {
```

Comment:
shall we move this to present, since it is not used anywhere else?

Reply:
should we use dict() or {}? we use dict() most of the time for dict instantiation, but not sure if it matters

Reply:
I'll keep it here for now. It keeps all the `module.params.get` in the same place.

There appears to be a difference in performance. Eg https://doughellmann.com/posts/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/
The conclusion looks like `{}` is more efficient. Although this post is for 2.7, I can still observe a similar result in 3.

Reply:
I think we have always used dict() to keep the code consistent with how we integrate it with Ansible.
Example:
```
argument_spec.update(
        template=dict(type="str", required=True, aliases=["fabric_template"]),
```
If we decide to go with {} for the payload then we would likely need to change it everywhere.